config/initializers/sidekiq.rb

```diff
@@ -12,6 +12,12 @@ def sanitize_redis_config(cfg)
   cfg.slice(*REDIS_PARAMS_WHITELIST)
 end
 
+class Sidekiq::Batch
+  # Override the default value of 30 days with 6 hours
+  remove_const(:BID_EXPIRE_TTL) if defined?(BID_EXPIRE_TTL)
```

Comment:
the `if` condition might mislead us in case load ordeing becomes broken and we run the initializer before the gem.

On the other hand even if so happens, that will not be a critical issue so better we start. idk whether we need this `if`

Reply:
We do, because if we run `remove_const` on a non-existing constant name, it will throw a
```
NameError: constant Whatever::BID_EXPIRE_TTL not defined
```

Reply:
But it should exists already or something very wrong must have happened. It will not existonly if gem was not already loaded. And it should have been loaded.

Because if gem is not loaded already, then we will set the value. When gem is loaded, then it will go back to defaults.

But as I said, something really bad should have happened that the gem was not loaded. Or not?

Reply:
> Because if gem is not loaded already, then we will set the value. When gem is loaded, then it will go back to defaults.

So summarizing: if we leave the `if`, a broken order will overwrite the value leaving a warning in the logs; on the other hand if we remove the `if`, a broken loading order would raise a `NameError` exception.

In that case it's probably better to leave the `if` so nothing crashes.

If you want to be sure, you can wrap this snippet under `to_prepare` to ensure everything is loaded before replacing the const:

```ruby
class Sidekiq::Batch
  Rails.application.config.to_prepare do
    # Override the default value of 30 days with 6 hours
    remove_const(:BID_EXPIRE_TTL) if defined?(BID_EXPIRE_TTL)
    const_set(:BID_EXPIRE_TTL, 21_600)
  end
end
```

Reply:
I just tried to wrap it under `Rails.application.config.to_prepare` and it crashed,so better don't do it... 🫠

Reply:
Gems should never be reloaded so wrapping under `to_prepare` doesn't seem necessary.

> So summarizing: if we leave the if, a broken order will overwrite the value leaving a warning in the logs; on the other hand if we remove the if, a broken loading order would raise a NameError exception.

Yeah, this is why I agree it to stay like this.

On the other hand, something has to be seriously broken for load order to differ. And maybe we want to make sure catching it as soon as it happens and never reach production. Because we have a lot in initializers that relies on gems being loaded first.